### ISSUE TITLE
Add support for ELF Core dumps

### ISSUE THREAD

**Issue by JaryDev** [fireeye/capa#784]
### Summary

Would it be possible to add support for ELF core dump files, as created by gcore, gdb etc. 

### Motivation

Ability to analyse dumped process memory on Linux. Helping to investigate code injection. 

### Describe alternatives you've considered

Currently capa provides an error message "Input file does not appear to target a supported OS" when trying to run it against a core dump. Would a solution be as trivial as changing the e_type field to ET_EXEC / ET_DYN in the executable header of a core file to allow capa to run?

**Comment by williballenthin** [comment]
We had a similar idea for analyzing Windows memory dumps; however, there's stumbling block: of all the loaded modules in the dump, which one should be analyzed and displayed? I can imagine a user interface that prompts the user for which module to inspect, but something like this is probably beyond the scope of the primary capa tool. Could we work together on an integration with capa by another tool, maybe like Volatility or Rekall

**Comment by JaryDev** [comment]
Good point made. FYI I stumbled across a quick work around yesterday using 'core2ELF64' which provides the option to rebuild either the ELF binary or any module independent from each other. These binaries then ran fine with capa. Perhaps this could be integrated in future? It essentially just iterates through the headers and dynamic segment to rebuild the binary, adjusting the program header table afterwards. 

IIRC Volatility only handles full dumps on Linux / VBox core dumps (in which the guest's physical mem is extracted), rather than per process core files? Going forwards it would indeed be cool have capa run across a full memory dump, but for the time being I'm just focussing on single processes.
